Feature/add contract to judy subsystem

jvdyfu opened the pull request:
the electrical contracts and test for armlifting subsystems

jvdyfu committed:
created contracts

jvdyfu committed:
created tests

jvdyfu committed:
finishing contracts and stuff

jvdyfu committed:
removed //

jvdyfu committed:
fixed tests

JohnGilb commented on src/main/java/competition/subsystems/SubsystemDefaultCommandMap.java:
```diff
@@ -8,7 +8,6 @@
 
 @Singleton
 public class SubsystemDefaultCommandMap {
-    // For setting the default commands on subsystems
```
No need to remove this comment.

JohnGilb commented on src/main/java/competition/subsystems/armlifting/CollectorArmLiftingSubsystem.java:
```diff
@@ -51,9 +59,15 @@ public boolean isAtMinimum() {
+        if(contract.isCollectorArmLiftingReady()) {
+            liftingCollectorArmMotor.simpleSet(power);
+        }
+
```
This needs to be after all the checks for maximum/minimum power, or it won't work.

JohnGilb commented on src/test/java/competition/subsystems/armlifting/CollectorArmLiftingSubsystemTest.java:
```diff
@@ -0,0 +1,17 @@
+    public void testCollectorArmLifting() {
+        CollectorArmLiftingSubsystem collectArmLifting = this.injector.getInstance(CollectorArmLiftingSubsystem.class);
+        collectArmLifting.up();
+        assertEquals(1, collectArmLifting.liftingCollectorArmMotor.getMotorOutputPercent(), 0.001);
```
you should test all methods!

JohnGilb commented on src/test/java/competition/subsystems/armlifting/FrontCollectingSubsystemTest.java:
```diff
@@ -0,0 +1,17 @@
+    public void testFrontCollecting() {
+        FrontCollectingSubsystem frontCollect = this.injector.getInstance(FrontCollectingSubsystem.class);
+        frontCollect.intake();
+        assertEquals(1, frontCollect.frontCollectingMotor.getMotorOutputPercent(), 0.001);
```
Test all methods!

jvdyfu committed:
created hanger

jvdyfu committed:
added hanger subsystem and electrical contract +test

jvdyfu committed:
hanger subsystem

jvdyfu committed:
added back comment on subsystemdefaultcommandmap

added more tests for frontcollectsubsystem and collectorarmliftingsubsystem

created test for hangersubsystem

JohnGilb commented on src/main/java/competition/subsystems/hanger/HangerSubsystem.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    public void grabClamp () {
+        setPower(1);
```
use Hanging Power here instead of `1`

JohnGilb commented on src/test/java/competition/subsystems/armlifting/CollectorArmLiftingSubsystemTest.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    @Test
+    public void testCollectorArmLiftingStop() {
```
This doesn't actually test that the system stops - you need to have it moving first to confirm that it will stop.

JohnGilb commented on src/test/java/competition/subsystems/armlifting/FrontCollectingSubsystemTest.java:
```diff
@@ -0,0 +1,34 @@
+
+    
+    @Test
+    public void testFrontCollectingStop() {
```
Same stop problem as above

JohnGilb commented on src/test/java/competition/subsystems/hanger/HangerSubsystemTest.java:
```diff
@@ -0,0 +1,45 @@
+    @Test
+    public void testHangerStop() {
+        HangerSubsystem hanger = this.injector.getInstance(HangerSubsystem.class);
+        hanger.stop();
```
same stop problem.

jvdyfu committed:
Merge branch 'master' into feature/add-contract-to-Judy-subsystem

# Conflicts:
#	src/main/java/competition/ActualElectricalContract.java
#	src/main/java/competition/subsystems/hood/HoodSubsystem.java
#	src/main/java/competition/subsystems/shooterwheel/commands/SpinningShooterWheelCommand.java

jvdyfu committed:
fixed

jvdyfu committed:
fixed stop tests

jvdyfu committed:
fixed build errors

JohnGilb commented on src/main/java/competition/subsystems/armlifting/CollectorArmLiftingSubsystem.java:
```diff
@@ -21,16 +23,23 @@
+        this.contract = contract;
+
+        if (contract.isCollectorArmLiftingReady()) {
+            this.liftingCollectorArmMotor = factory.createCANTalon(contract.liftingCollectorArmMotor().channel);
```
you also need to set inversion, if any.

JohnGilb commented on src/main/java/competition/subsystems/armlifting/FrontCollectingSubsystem.java:
```diff
@@ -3,43 +3,60 @@
-        // make a stop for both subsystems as the default
-
+        if (contract.isFrontCollectingReady()) {
+            this.frontCollectingMotor = factory.createCANTalon(contract.frontCollectingMotor().channel);
```
inversion, if any

JohnGilb commented on src/test/java/competition/subsystems/hanger/HangerSubsystemTest.java:
```diff
@@ -0,0 +1,49 @@
+    @Test
+    public void testHangerStop() {
+        HangerSubsystem hanger = this.injector.getInstance(HangerSubsystem.class);
+        hanger.extendHanger();
```
Basic pattern for this thing:
1) Make it extend
2) Verify (via assert) it is extending
3) Make it stop
4) Verify (via assert) it has stopped.

jvdyfu committed:
setted inversion and fixed the tests for stop

jvdyfu committed:
Merge branch 'master' into feature/add-contract-to-Judy-subsystem
